src/ecc_key.c

```diff
@@ -1280,6 +1284,64 @@ int ec_key_sec2prot(struct zpc_ec_key *ec_key, enum ec_key_sec sec)
 	return 0;
 }
 
+int ec_key_clr2prot(struct zpc_ec_key *ec_key, const unsigned char *privkey,
+					unsigned int privlen)
+{
+	struct pkey_kblob2pkey3 io;
+	unsigned char buf[sizeof(struct clearkeytoken) + 80];
+	struct clearkeytoken *clrtok = (struct clearkeytoken *)&buf;
+	int rc;
+
+	memset(buf, 0, sizeof(buf));
+	clrtok->version = 0x02;
+	clrtok->keytype = curve2pkey_keytype[ec_key->curve];
+	switch (clrtok->keytype) {
+	case PKEY_KEYTYPE_ECC_P256:
+	case PKEY_KEYTYPE_ECC_P384:
+	case PKEY_KEYTYPE_ECC_ED25519:
+		memcpy(clrtok->clearkey, privkey, privlen);
+		clrtok->len = privlen;
+		break;
+	case PKEY_KEYTYPE_ECC_P521:
+		memcpy(clrtok->clearkey + 80 - privlen, privkey, privlen);
```

Comment:
sorry, I didn't read the specs, but where do this + 80 and + 64 come from?
And isn't the default case of this switch just PKEY_KEYTYPE_ECC_ED448

Reply:
Yes, the CPACF key buffer for p521 is in fact 80 bytes and not just 66. And for ED448 it's 64 bytes and not 57. The key material inside the buffers is right aligned. 
Yes, the default case is PKEY_KEYTYPE_ECC_ED448. So we could also make a case for it and leave default out. Other values than the 5 known ones cannot show up.